### PR TITLE
@/commons-ui/core/Section

### LINKED ISSUE #30: @commons-ui/core/Section
A (page) Section component that uses [`@mui/material/Container`](https://mui.com/components/container/) and `theme.margins` from #29 to compute max-width at current breakpoint.

Depends on #29

### PR DESCRIPTION
**Pull request by gertie-sheshe**
## Description

Resolves #30 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

Peek 2022-04-13 11-43

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by kilemensi — COMMENTED** [review]
👍🏽 

----

As I indicated in the issues (#30), we need to use [`@mui/material/Container`](https://mui.com/components/container/). This is because:

1. `Container` completely replaces `Layout` i.e. there is nothing that `Layout` offers that `Container` doesn't.
2. `Section` will be `<Container fixed ...>` but using the `contentWidths` we've introduced in #29 instead of [`maxWidth`](https://github.com/mui/material-ui/blob/c406cd6f89dcab6a223dbe77d833bc947ff64fdb/packages/mui-material/src/Container/Container.js#L54-L78).

**Review by nerdlet — APPROVED** [review]
@gertie-sheshe Looks good 🚀 🚀 

I think `Section` is missing here 👇🏽 
 https://github.com/CodeForAfrica/ui/blob/chore/section/packages/commons-ui-core/src/index.js

**Review by KhadijaMahanga — COMMENTED** [review]
@gertie-sheshe shouldn't we be overriding the `Container`'s fixed [css](react-multi-carousel) prop? (i.e fixed  width)

File: packages/commons-ui-core/src/Section/index.js [hunk 1]
```diff
@@ -1,16 +1,17 @@
 import { styled } from "@mui/material/styles";
+import { Container } from "@mui/material";
 
-const Section = styled("div")(({ theme }) => ({
+const ContainerStyled = styled(Container)(({ theme }) => ({
```
Comment (kilemensi) [inline packages/commons-ui-core/src/Section/index.js:4]:
```suggestion
const SectionRoot = styled(Container)(({ theme }) => ({
```

Sticking to MUI naming

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

We only want to override the `maxWidth` when it's `fixed` (default for us). `Container` uses breakpoints to set `maxWidth` but we want to use `contentWidths`. Essentially changing [this](https://github.com/mui/material-ui/blob/e39ba7a0ef3afe1c44c04cf7b4d57f894cf525f7/packages/mui-material/src/Container/Container.js#L54-L65) part of the code only.

**Review by KhadijaMahanga — APPROVED** [review]
🚀

**Pull request merged**